### PR TITLE
Update TSIP VER to 1.15 for RX65N and RX72N

### PR DESCRIPTION
**Pull request by miyazakh**
Verified TSIP VER 1.15 on RX65N and RX72N

The PR also includes:
 fix -140 error when enabling -O2 CC-RX compiler optimization option
 fix no_crypto_hash macro works
 fix a compile error by clang

# Description

# Testing

Run Crypt, Benchmark and Client on those board

# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: IDE/Renesas/e2studio/RX72N/EnvisionKit/wolfssl_demo/user_settings.h [hunk 1]
```diff
@@ -60,8 +61,10 @@
 #define FREERTOS
 #define FREERTOS_TCP
 
-
-
+#if !defined(FREERTOS_TCP)
+ #define WOLFSSL_NO_SOCK
```
Comment (cconlon) [inline IDE/Renesas/e2studio/RX72N/EnvisionKit/wolfssl_demo/user_settings.h:65]: Indent should be 4 spaces here and in line below.

**Review by cconlon — CHANGES_REQUESTED** [review]
Changes look good, thanks Hide!  Just one minor comment about indentation spacing.

File: IDE/Renesas/e2studio/RX72N/EnvisionKit/wolfssl_demo/user_settings.h (hunk 1 above)
Comment (miyazakh) [reply IDE/Renesas/e2studio/RX72N/EnvisionKit/wolfssl_demo/user_settings.h]: Thank you for pointing it out. Fixed

In addition to that, I update Copyright year

**Comment by miyazakh** [comment]
Want to update one more file to fix compile error by clang

**Pull request merged**